src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -127,17 +127,13 @@ public Guid(int a, short b, short c, byte[] d)
                 throw new ArgumentException(SR.Format(SR.Arg_GuidArrayCtor, "8"), nameof(d));
             }
 
+            Unsafe.SkipInit(out this);
```

Comment:
Does this even do anything? Aren’t objects zero-initialized before constructors are invoked?

Reply:
> We're reluctant to add more use of unsafe without good reason. @EgorBo do we have anything written down?

This seems reasonable in such a simple constructor. `Guid` already has extensive `Unsafe` usage:

https://github.com/dotnet/runtime/blob/7dd68f40ab9ae6f631d79b76f2d9b464f7ecee36/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L244-L247

https://github.com/dotnet/runtime/blob/7dd68f40ab9ae6f631d79b76f2d9b464f7ecee36/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L988-L993

Reply:
> Does this even do anything? Aren’t objects zero-initialized before constructors are invoked?

In the case of structs, I believe not. The purpose of `Unsafe.SkipInit` is to avoid initial zero-initialization in the ctor.

Reply:
> @EgorBo do we have anything written down?

Not yet.
I would be surprised if anyone uses this overload at all, so any unsafe hacks look unnecessary here.

>  Guid already has extensive Unsafe usage:

I don't think it's a good motivation to add more. I think these copies could be optimized by jit if we revive https://github.com/dotnet/runtime/pull/95823

Reply:
Yep, this overload is rare, reverted use of `Unsafe`, and opened https://github.com/dotnet/runtime/issues/105257.